src/input-select/input-range-select.tsx

```diff
@@ -0,0 +1,316 @@
+import React, { useEffect, useRef, useState } from "react";
+import { ArrowRight, IndicateBar } from "src/date-input/date-input.style";
+import { DropdownList } from "../shared/dropdown-list/dropdown-list";
+import { StringHelper } from "../util/string-helper";
+import { InputSelectWrapper } from "./input-select-wrapper";
+import {
+    Divider,
+    LabelContainer,
+    PlaceholderLabel,
+    RangeIcon,
+    Selector,
+    ValueLabel,
+} from "./input-select.styles";
+import { InputRangeProp, InputRangeSelectProps } from "./types";
+
+type RangeType = "from" | "to";
+
+export const InputRangeSelect = <T, V>({
+    selectedOptions,
+    placeholders = { from: "Select", to: "Select" },
+    options,
+    disabled,
+    readOnly,
+    error,
+    "data-testid": testId,
+    id,
+    enableSearch = false,
+    searchFunction,
+    searchPlaceholder,
+    valueExtractor,
+    valueToStringFunction,
+    listExtractor,
+    displayValueExtractor,
+    onSelectOption,
+    listStyleWidth,
+    onShowOptions,
+    onHideOptions,
+    onRetry,
+    optionsLoadState = { from: "success", to: "success" },
+    optionTruncationType = "middle",
+    renderCustomSelectedOption,
+    renderListItem,
+    renderCustomCallToAction,
+    ...otherProps
+}: InputRangeSelectProps<T, V>): JSX.Element => {
+    // =============================================================================
+    // CONST, STATE
+    // =============================================================================
+    const [selected, setSelected] = useState<InputRangeProp<T>>({
+        from: selectedOptions?.from,
+        to: selectedOptions?.to,
+    });
+    const [showOptions, setShowOptions] = useState<InputRangeProp<boolean>>({
```

Comment:
Actually we can reuse `currentElementType` state to determine what to show. A singular state works since the from and to options will never appear at the same time. Furthermore i see them being set together. 

Perhaps we can rename the state to be `focusedInput` and use `RangeType | "none"`

I would recommend standardising the terminology for easier understanding also. Since you are using `from` and `to`, let's use this throughout the component.

Reply:
Done